server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -1198,6 +1198,12 @@ public void getLocalShuffleData(
                       offset,
                       length,
                       storageId);
+          reply =
+              GetLocalShuffleDataResponse.newBuilder()
+                  .setStatus(status.toProto())
+                  .setRetMsg(msg)
+                  .setData(UnsafeByteOperations.unsafeWrap(sdr.getData()))
```

Comment:
the operation of `sdr.getData()` will cost too much time, so it should be scoped into the metrics recording

Reply:
Got it.